src/renderer/components/Settings/index.js

```diff
@@ -0,0 +1,139 @@
+// @ts-check
+import * as React from 'react'
+import { SettingsMenu } from './SettingsMenu'
+import { defineMessages } from 'react-intl'
+import MapIcon from '@material-ui/icons/MapOutlined'
+import InfoIcon from '@material-ui/icons/InfoOutlined'
+import { BackgroundMaps } from './BackgroundMaps'
+import { makeStyles } from '@material-ui/core'
+import Fade from '@material-ui/core/Fade'
+import Paper from '@material-ui/core/Paper'
+import { AboutMapeo } from './AboutMapeo'
+
+const m = defineMessages({
+  backgroundMap: 'Background Map',
+  aboutMapeo: 'About Mapeo'
+})
+
+const FADE_DURATION = 700
+
+/** @typedef {'BackgroundMap' | 'AboutMapeo'} SettingTabId */
+
+/** @type {import('./SettingsMenu').SettingsTabs[]} */
+const tabs = /** @typedef {const} */ [
+  {
+    tabId: 'BackgroundMap',
+    icon: <MapIcon />,
+    label: m.backgroundMap
+  },
+  {
+    tabId: 'AboutMapeo',
+    icon: <InfoIcon />,
+    label: m.aboutMapeo
+  }
+]
+
+/**
+ *
+ * @typedef SettingsProp
+ * @prop {boolean} reset
+ * @prop {function(boolean):void} setReset
+ * @prop {boolean} fadeIn
+ */
+
+/** @param {SettingsProp} props */
+export const Settings = ({ reset, setReset, fadeIn }) => {
+  const [menuVisible, setMenuVisibility] = React.useState(true)
+
+  /** @type {import('./SettingsMenu').SettingsTabs['tabId'] | false} */
+  const initialState = /** {const} */ (false)
```

Comment:
would make this variable a little more descriptive or just inline it to the useState initializer.

Reply:
This is actually a very annoying this about jsdoc. The typescript equivalent would be `useState<SettingsTabs['tabId']|false>` In order to do this with JSdoc, i need to initialize the first state of the `useState()` above it as I did here.

Reply:
I renamed it so it is a little clearer: https://github.com/digidem/mapeo-desktop/pull/678/commits/73472a2872649cdd5aeff1f3772395226c25a63d